### PR TITLE
adds a mailmap to make the logs more clear

### PR DESCRIPTION
**Pull request by lewisc**
This is a minor but nice feature that makes reading certain git logs noticeably easier. It also allows for email changes without doing a rebase (which is significant if someone's contact email changes, obviously).

### CODE REVIEW

File: .mailmap [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Mailmap to correct users
+
+Joseph K. Bradley <joseph.kurata.bradley@gmail.com> jkbradley <joseph.kurata.bradley@gmail.com>
+Xiangrui Meng <meng@databricks.com>
```
Comment (mengxr) [inline .mailmap]: It seems that we don't need this line because I didn't use other names or emails. Same for L5, L10, L11, L12.

File: .mailmap [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Feynman Liang <fliang@databricks.com>
+Felix Cheung <felixcheung_m@hotmail.com> Felix Cheung <fcheung@avvo.com>
+Felix Cheung <felixcheung_m@hotmail.com> felixcheung <felixcheung_m@hotmail.com>
+Felix Cheung <felixcheung_m@hotmail.com> Felix Cheung <fcheung@avvo.com>
```
Comment (mengxr) [inline .mailmap]: Is it the same as L6?

File: .mailmap [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+Felix Cheung <felixcheung_m@hotmail.com> Felix Cheung <felixcheung@users.noreply.github.com>
+Qingpeng Zhang <qingpeng@msu.edu>
+Reynold Xin <rxin@databricks.com>
+Bill Chambers <bill@databricks.com>
+Shagun <sshagunsodhani@gmail.com>
```
Comment (mengxr) [inline .mailmap]: `Shagun Sodhani <sshagunsodhani@gmail.com> Shagun <sshagunsodhani@gmail.com>`

Ref: https://github.com/shagunsodhani

File: .mailmap [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+fred <fred@calculquebec.ca>
```
Comment (mengxr) [inline .mailmap]: `Frederick Lefebvre <fred@calculquebec.ca> fred <fred@calculquebec.ca>`

Ref: https://github.com/fred-lefebvre

File: .mailmap [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# Mailmap to correct users
+
+Joseph K. Bradley <joseph.kurata.bradley@gmail.com> jkbradley <joseph.kurata.bradley@gmail.com>
+Xiangrui Meng <meng@databricks.com>
+Feynman Liang <fliang@databricks.com>
+Felix Cheung <felixcheung_m@hotmail.com> Felix Cheung <fcheung@avvo.com>
+Felix Cheung <felixcheung_m@hotmail.com> felixcheung <felixcheung_m@hotmail.com>
```
Comment (felixcheung) [inline .mailmap]: you could remove all of mine except L7

**Comment by lewisc** [comment]
I have responded to your review updates requests, thanks.

**Comment by lewisc** [comment]
In order to assist with the discussion, this is what I am cleaning up

before


after

**Comment by mengxr** [comment]
Merged. Thanks!

**Pull request merged**